src/nets/e2e_asr_attctc_th.py

```diff
@@ -304,35 +394,36 @@ def reset(self):
         self.enc_h = None
         self.pre_compute_enc_h = None
 
-    def forward(self, enc_hs, dec_z, scaling=2.0):
+    def forward(self, enc_hs_pad, enc_hs_len, dec_z, att_prev, scaling=2.0):
         '''
 
         :param enc_hs:
         :param dec_z:
         :param scaling:
         :return:
         '''
-        batch = len(enc_hs)
+        batch = enc_hs_pad.size(0)
         # pre-compute all h outside the decoder loop
         if self.pre_compute_enc_h is None:
-            self.enc_h = F.pad_sequence(enc_hs)  # utt x frame x hdim
+            self.enc_h = enc_hs_pad  # utt x frame x hdim
             self.h_length = self.enc_h.shape[1]
             # utt x frame x att_dim
-            self.pre_compute_enc_h = F.tanh(linear_tensor(self.mlp_enc, self.enc_h))
+            self.pre_compute_enc_h = linear_tensor(self.mlp_enc, self.enc_h)
```

Comment:
Does it need to perform `tanh`?

Reply:
Hmm, I'm not sure why I delete it. However I will follow the reference  impl of chainer.

Reply:
In eq 9 in https://arxiv.org/pdf/1609.06773.pdf, `pre_compute_h` corresponds to `V h_l ` that does not have tanh. Anyway it might not be big deal because we cannot see big difference in https://github.com/espnet/espnet/issues/9#issuecomment-353366442